vnext/Microsoft.ReactNative/IReactDispatcher.cpp

```diff
@@ -29,20 +29,35 @@ void ReactDispatcher::Post(ReactDispatcherCallback const &callback) noexcept {
 }
 
 /*static*/ IReactDispatcher ReactDispatcher::UIThreadDispatcher() noexcept {
-  return make<ReactDispatcher>(Mso::DispatchQueue::MakeCurrentThreadUIQueue());
-}
-
-/*static*/ ReactPropertyId<IReactDispatcher> ReactDispatcher::UIDispatcherProperty() noexcept {
-  static ReactPropertyId<IReactDispatcher> uiThreadDispatcherProperty{L"ReactNative.Dispatcher", L"UIDispatcher"};
+  static thread_local weak_ref<IReactDispatcher> *tlsWeakDispatcher{nullptr};
+  IReactDispatcher dispatcher;
```

Comment:
>dispatcher [](start = 19, length = 10)

uninitalized memory if queue is null? Or did I miss a new c++/winrt feature?

Reply:
The interface-like looking types in C++/WinRT are really specialized com_ptr smart pointers. This line calls the default constructor of the IReactDispatcher that sets it to nullptr. I can make it more explicit to avoid any confusion.

---
In reply to: [430590699](https://github.com/microsoft/react-native-windows/pull/5007#discussion_r430590699) [](ancestors = 430590699)

Reply:
doh, sorry... thanks!

---
In reply to: [430592838](https://github.com/microsoft/react-native-windows/pull/5007#discussion_r430592838) [](ancestors = 430592838,430590699)

Reply:
Added passing nullptr to constructor to make it explicit.

---
In reply to: [430593326](https://github.com/microsoft/react-native-windows/pull/5007#discussion_r430593326) [](ancestors = 430593326,430592838,430590699)